### PR TITLE
 [MaterialDividerItemDecoration] When lastItemDecorated false then creates a blink with the last view contains a dropdown from visible to gone

### PR DESCRIPTION
**Pull request by mnlmato**
solves #2771

### CODE REVIEW

**Review by NTPaa — APPROVED** [review]

File: lib/java/com/google/android/material/divider/MaterialDividerItemDecoration.java [hunk 1]
```diff
@@ -364,6 +364,16 @@ private void drawForHorizontalOrientation(@NonNull Canvas canvas, @NonNull Recyc
     canvas.restore();
   }
 
+  private int getDividerCount(RecyclerView recyclerView) {
```
Comment (drchen) [inline lib/java/com/google/android/material/divider/MaterialDividerItemDecoration.java:367]: getItemCount

File: lib/java/com/google/android/material/divider/MaterialDividerItemDecoration.java [hunk 2]
```diff
@@ -364,6 +364,16 @@ private void drawForHorizontalOrientation(@NonNull Canvas canvas, @NonNull Recyc
...
+    int dividerCount;
```
Comment (drchen) [inline lib/java/com/google/android/material/divider/MaterialDividerItemDecoration.java:368]: itemCount

File: lib/java/com/google/android/material/divider/MaterialDividerItemDecoration.java [hunk 3]
```diff
@@ -364,6 +364,16 @@ private void drawForHorizontalOrientation(@NonNull Canvas canvas, @NonNull Recyc
...
+    if (recyclerView.getAdapter() != null) {
+      dividerCount = recyclerView.getAdapter().getItemCount();
+    } else {
+      dividerCount = EMPTY_ITEMS_RECYCLER_VIEW;
+    }
+    return dividerCount;
```
Comment (drchen) [inline lib/java/com/google/android/material/divider/MaterialDividerItemDecoration.java:374]: Can be simplified to:

return recyclerView.getAdapter() == null ? 0 : recyclerView.getAdapter().getItemCount();

File: lib/java/com/google/android/material/divider/MaterialDividerItemDecoration.java [hunk 4]
```diff
@@ -59,6 +59,7 @@ public class MaterialDividerItemDecoration extends ItemDecoration {
   public static final int VERTICAL = LinearLayout.VERTICAL;
 
   private static final int DEF_STYLE_RES = R.style.Widget_MaterialComponents_MaterialDivider;
+  private static final int EMPTY_ITEMS_RECYCLER_VIEW = 0;
```
Comment (drchen) [inline lib/java/com/google/android/material/divider/MaterialDividerItemDecoration.java:62]: No need to declare a constant for 0. It's usage is self-explanatory.

**Review by drchen — APPROVED** [review]

**Pull request closed without merging**